src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioWindowsLanguageServerFeatureOptions.cs

```diff
@@ -51,6 +53,13 @@ public VisualStudioWindowsLanguageServerFeatureOptions(LSPEditorFeatureDetector
             var usePreciseSemanticTokenRanges = featureFlags.IsFeatureEnabled(UsePreciseSemanticTokenRangesFeatureFlag, defaultValue: false);
             return usePreciseSemanticTokenRanges;
         });
+
+        _useRoslynAnalyzerDiagnostics = new Lazy<bool>(() =>
+        {
+            var featureFlags = (IVsFeatureFlags)AsyncPackage.GetGlobalService(typeof(SVsFeatureFlags));
+            var usePreciseSemanticTokenRanges = featureFlags.IsFeatureEnabled(UseRoslynAnalyzerDiagnosticsFeatureFlag, defaultValue: true);
```

Comment:
nit: seem to be using outdated variable name `usePreciseSemanticTokenRanges` from copy paste